test/parallel/test-child-process-exec-kill-throws.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+// Flags: --expose_internals
+const common = require('../common');
+const assert = require('assert');
+const cp = require('child_process');
+const internalCp = require('internal/child_process');
+
+if (process.argv[2] === 'child') {
+  // Keep the process alive and printing to stdout.
+  setInterval(() => { console.log('foo'); }, 1);
+} else {
+  // Monkey path ChildProcess#kill() to kill the process and then throw.
+  const kill = internalCp.ChildProcess.prototype.kill;
+
+  internalCp.ChildProcess.prototype.kill = function() {
+    kill.apply(this, arguments);
+    throw new Error('mock error');
+  };
+
+  const cmd = `${process.execPath} ${__filename} child`;
+  const options = { maxBuffer: 0 };
+  const child = cp.exec(cmd, options, common.mustCall((err, stdout, stderr) => {
+    // Verify that if ChildProcess#kill() throws, the error is reported.
```

Comment:
node could decided to kill the child, but between when it decided and the child is killed, the child can exit. It would be possible to special  case https://github.com/nodejs/node/blob/3268863ebc40d1f0beee61b044c492b43fa57fa5/lib/child_process.js#L265 for `ESRCH`, and not call exit handler in that case, because it just means that node has not yet epolled the exit status, but it will in the next tick or so. But the event ordering of the timeout vs the process exit is indeterminate by nature, so its no big deal.